pkg/render/render_test.go

```diff
@@ -0,0 +1,114 @@
+package render
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("RenderTemplate", func() {
+	Context("when a valid YAML manifest is given as a parameter", func() {
+		It("should successfully render", func() {
+			renderData := MakeRenderData()
+
+			renderedFromYAML, err := RenderTemplate("testdata/simple.yaml", &renderData)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(renderedFromYAML).To(HaveLen(1))
+			expected := `
+{
+	"apiVersion": "v1",
+	"kind": "Pod",
+	"metadata": {
+		"name": "busybox1",
+		"namespace": "ns"
+	},
+	"spec": {
+		"containers": [
+			{
+  				"image": "busybox"
+			}
+		]
+	}
+}
+`
+			Expect(renderedFromYAML[0].MarshalJSON()).To(MatchJSON(expected))
+		})
+	})
+
+	Context("when a identical YAML and JSONs manifest are given as a parameters", func() {
+		It("should return identical objects", func() {
+			renderData := MakeRenderData()
+
+			renderedFromYAML, err := RenderTemplate("testdata/simple.yaml", &renderData)
+			Expect(err).NotTo(HaveOccurred())
+
+			renderedFromJSON, err := RenderTemplate("testdata/simple.json", &renderData)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(renderedFromJSON).To(Equal(renderedFromYAML))
+		})
+	})
+
+	Context("when YAML manifest with multiple objects given as a parameter", func() {
+		It("should successfully render list of objects", func() {
+			renderData := MakeRenderData()
+
+			rendered, err := RenderTemplate("testdata/multiple.yaml", &renderData)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(rendered).To(HaveLen(3))
+
+			Expect(rendered[0].GetObjectKind().GroupVersionKind().String()).To(Equal("/v1, Kind=Pod"))
```

Comment:
Order here is not relevant also I suppose we need to compare with all the fields not just GroupVersionKind

Reply:
The rest of the fields is not important, we test that in previous (single pod) tests. This one is just to make sure all objects are there.

Reply:
You are right that the order is not important, but I'd say it is not worth making the test know it. Given test files is static and part of the test. If you think there is a benefit in making it more robust, tell me.